### DISCUSSION TITLE
Invoke method throws: Jint.Runtime.JavaScriptException: 'Can only invoke functions' exception

### DISCUSSION THREAD

**Discussion by aglasencnik** [sebastienros/jint#1679] [Q&A]
Hi, I have the following code:
```csharp
var code = File.ReadAllText("emmet.min.js");

var engine = new Engine()
    .Execute(code);

var input = "div>ul>li*5>a";
var output = engine.Evaluate("emmet.expandAbbreviation('div>ul>li*5>a')").ToObject();
var output2 = engine.Invoke("emmet.expandAbbreviation", input).ToString();

Console.WriteLine($"Input: {input}");
Console.WriteLine($"Output: {output}");
Console.ReadKey();
```
output variable returns the correct result, but the output2 throws the: Jint.Runtime.JavaScriptException: 'Can only invoke functions' exception
I basically compile a typescript npm package and load the js minified version here and execute it... After I want to invoke one of the methods and get the result...

**Comment by lahma** [comment]
Would you like to share full repro via public repository and maybe even the Jint version you are using?

**Reply by aglasencnik** [reply]
Hi, I don't have a repository, since this was just a test, but I can write all the steps I took (I was just testing Jint)... Also I don't really have experience with npm or javascript so it is possible I messed it up there...
Ok so I have installed the npm package in some empty folder and have this index.js:
```javascript
export {
    default as expandAbbreviation,
    CSSAbbreviationScope,
    extract,
    markup,
    markupAbbreviation,
    parseMarkup,
    parseStylesheet,
    parseStylesheetSnippets,
    resolveConfig,
    stringifyMarkup,
    stringifyStylesheet,
    stylesheet,
    stylesheetAbbreviation
} from 'emmet';
```
And this in the webpack.config.js:
```javascript
const path = require('path');
const TerserPlugin = require('terser-webpack-plugin');

module.exports = {
    entry: './src/index.js',
    output: {
        filename: 'emmet.min.js',
        path: path.resolve(__dirname, 'dist'),
        library: 'emmet',
        libraryTarget: 'umd',
        globalObject: 'this'
    },
    module: {
        rules: [
            {
                test: /\.js$/,
                exclude: /node_modules/,
                use: {
                    loader: 'babel-loader',
                    options: {
                        presets: ['@babel/preset-env']
                    }
                }
            }
        ]
    },
    optimization: {
        minimize: true,
        minimizer: [new TerserPlugin()],
    }
};
```
Then I run: `npx webpack --config webpack.config.js`, and copy the .min.js file in VS and set the copy to output to always...
Then I have the code I pasted earlier....

**Reply by aglasencnik** [reply]
Oh also, I am using the most recent Jint version of 3.0 beta

**Reply by aglasencnik** [reply]
I wanted to later have a service class where I would have a 1 readonly instance of the Engine, and then invoke those functions
